benchmark/framework/cli.py

```diff
@@ -0,0 +1,150 @@
+
+
+class TestCli:
+    """Performance test client."""
+
+    def __init__(
+        self,
+        func: Callable,
+        executor_class=Executor,
+        report_printer_class=PerformanceReportPrinter,
+    ):
+        """
+        Make performance client.
+
+        func - should be function with first parameter multithreading.Queue.
+        func  - should have docstring, and default values for every extra argument.
+        """
+        self.func_details = FuncDetails(func)
+        self.func_details.check()
+        self.func = func
+        self.executor_class = Executor
+        self.report_printer_class = report_printer_class
+
+    def _make_command(self) -> Command:
+        """Make  cli.core.Command."""
+        return Command(
+            None,  # type: ignore
+            params=self._make_command_params(),
+            callback=self._callback,
+            help=self._make_help(),
+        )
+
+    def _make_command_params(self) -> Optional[List[Parameter]]:
+        """Make parameters and arguments for cli.Command."""
+        return list(self.executor_params().values()) + [self._function_args()]
+
+    def _make_help(self) -> str:
+        """Make help for command."""
+        return inspect.cleandoc(
+            f"""
+        {self.func_details.doc}
+
+        ARGS is function arguments in format: `{','.join(self.func_details.argument_names)}`
+
+        default ARGS is `{self.func_details.default_argument_values_as_string}`
+        """
+        )
+
+    def executor_params(self) -> Dict[str, Parameter]:
+        """Get parameters used by Executor."""
+        return {
+            "timeout": Option(
+                ["--timeout"],
+                default=10,
+                show_default=True,
+                help="Executor timeout in seconds",
+            ),
+            "period": Option(
+                ["--period"],
+                default=0.1,
+                show_default=True,
+                help="Period for measurement",
+            ),
+        }
+
+    def _function_args(self) -> Parameter:
+        """Get arguments requireed by test function."""
+        return DefaultArgumentsMultiple(
+            ["args"], default=[self.func_details.default_argument_values_as_string],
+        )
+
+    def run(self) -> None:
+        """Run performance test."""
+        command = self._make_command()
+        command()
+
+    def _callback(self, **exec_params) -> None:
+        """Run test on command."""
+        args_list = exec_params.pop("args")
+
+        executor = self.executor_class(**exec_params)
+        exec_reports = []
+
+        for args in args_list:
+            exec_report = executor.run(self.func, self._parse_arg_str(args))
+            exec_reports.append(exec_report)
+            report_printer = self.report_printer_class(
+                self.func_details, exec_params, [exec_report]
+            )
+            report_printer.print_()
+
+    def _parse_arg_str(self, args: str) -> tuple:
```

Comment:
the return type should be `Tuple` and declare what types are valid in it, e.g. `Tuple[str]`

Reply:
done